tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -3851,3 +3851,146 @@ async def test_dict_transformer_annotated_type():
+
+        key = TypeEngine.make_key(python_val, python_type)
+        assert key is not None
+        assert key in TypeEngine._CACHE
+
+        # Second call with same DataFrame
+        literal2 = TypeEngine.to_literal(ctx, python_val, python_type, expected)
+
+        # Verify async_to_literal was only called once
+        assert mock_async_to_literal.call_count == 1
+
+        assert literal1 is literal2
+
+    # Test with different data - should not use cache
+    different_val = [2, 1, 3, 4, 5]
+    literal3 = TypeEngine.to_literal(ctx, different_val, python_type, expected)
+    key_different = TypeEngine.make_key(different_val, python_type)
+
+    assert key_different is not key
+    assert key_different is not None
+    assert key_different in TypeEngine._CACHE
+
+    # Verify different literals are different objects
+    assert literal1 is not literal3
+
+    # Add many different values to test cache size limit
+    for i in range(200):  # More than the default maxsize of 128
+        test_val = [i, i+1, i+2]
+        test_type = typing.List[int]
+        test_expected = TypeEngine.to_literal_type(test_type)
+        TypeEngine.to_literal(ctx, test_val, test_type, test_expected)
+
+    # Cache should not exceed maxsize
+    assert len(TypeEngine._CACHE) == 128
+
+def test_type_engine_cache_with_dict():
+    ctx = FlyteContext.current_context()
+    python_val = {"a": [1, 2, 3]}
+    python_type = typing.Dict[str, typing.List[int]]
+    expected = TypeEngine.to_literal_type(python_type)
+    dict_transformer = TypeEngine.get_transformer(typing.Dict[str, typing.List[int]])
+    with mock.patch.object(dict_transformer, 'async_to_literal',
+                          wraps=dict_transformer.async_to_literal) as mock_async_to_literal:
+
+        # First call
+        literal1 = TypeEngine.to_literal(ctx, python_val, python_type, expected)
+
+        key = TypeEngine.make_key(python_val, python_type)
+        assert key is not None
+        assert key in TypeEngine._CACHE
+
+        # Second call with same DataFrame
+        literal2 = TypeEngine.to_literal(ctx, python_val, python_type, expected)
+
+        # Verify async_to_literal was only called once
+        assert mock_async_to_literal.call_count == 1
+
+        assert literal1 is literal2
+
+def test_make_key_with_annotated_types():
+    # Test with Annotated type
+    annotated_val = [1, 2, 3]
+    annotated_type = typing.Annotated[typing.List[int], "test_annotation"]
+
+    key = TypeEngine.make_key(annotated_val, annotated_type)
+    key_without_annotation = TypeEngine.make_key(annotated_val, typing.List[int])
+    # Should handle Annotated types correctly
+    assert key is not None
+    assert key_without_annotation is not None
+    assert key != key_without_annotation
+
+@pytest.mark.skipif("pandas" not in sys.modules, reason="Pandas is not installed.")
+def test_type_engine_cache_with_pandas():
+    import pandas as pd
```

Comment:
```suggestion
def test_type_engine_cache_with_pandas():
    pd = pytest.importorskip("pandas")
```

Instead of using `mark.skipif`, let's use `importorskip` to align with the usage in other files https://github.com/flyteorg/flytekit/blob/d03c14ce7c6748b01f0b3cf2450c469c3bc0327c/tests/flytekit/unit/types/structured_dataset/test_structured_dataset.py#L35